### DISCUSSION TITLE
Set automatically Image sizes equal to size of the container with layout "responsive"

### DISCUSSION THREAD

**Discussion by timuric** [vercel/next.js#22331] [Ideas]
**Describe the feature you'd like to request**

Most images on the web are responsive today, current implementation of the image component requires cumbersome to write ```sizes``` property. However when ```responsive``` image is ```lazy``` loaded Image container can calculate the size automatically. 

**Describe the solution you'd like**

When ```layout=responsive```  and ```loading=lazy``` set ```sizes``` property to according to the viewport width 

**Describe alternatives you've considered**

Here is a very basic pseudo code if you will
```
const Img = () => {
   const wrapperEl = useRef();
   const [sizes, setWrapperSize] = useState(null);

   useEffect(() => {
      const el = wrapperEl.current as HTMLDivElement;
      setWrapperSize(Math.round((el.offsetWidth / window.innerWidth) * 100));
   }, []);

   return (
      <div ref={wrapperEl}>
         <Image width={100} height={100} layout={"responsive"} src={"example.jpg"} sizes={sizes} />
      </div>
   );
};
```

**Comment by renanyudimt** [comment]
Yeah, im facing this problem too, in my case i have lot of images with different sizes, its terrible to setup different sizes for each image, what i did to work around it was wrap the image with `div` and through class set it width and height, then, apply `layout="fill"`

in example:

```
<div className="w-130px h-130px relative">
   <NextImage 
      src="/icon/icon.svg",
      layout="fill"
   />
</div>
```

then i can make it "response" with classes
```
<div className="h-70px w-70px md:h-100px md:w-100px lg:w-130px lg:h-130px relative">
   <NextImage 
      src="/icon/icon.svg",
      layout="fill"
   />
</div>
```

I did not figured out another better way to deal with it, if you do, please tell me then (:

PS: this example is using TailwindCSS with custom classes
